assignments/callbacks.js

```diff
@@ -1,55 +1,162 @@
-// Create a higher order function and invoke the callback function to test your work. You have been provided an example of a problem and a solution to see how this works with our items array.  Study both the problem and the solution to figure out the rest of the problems.
+/*
+  Create a higher order function and invoke the callback function to test your work. 
+  You have been provided an example of a problem and a solution to see how this works with our items array.  
+  Study both the problem and the solution to figure out the rest of the problems.
+*/
 
-const items = ['Pencil', 'Notebook', 'yo-yo', 'Gum'];
+const items = ['Pencil', 'Notebook', 'yo-yo', 'Gum', 'Notebook', 'yo-yo', 'Gum'];
 
-/* 
 
-  //Given this problem: 
-  
-  function firstItem(arr, cb) {
-    // firstItem passes the first item of the given array to the callback function.
-  }
+console.log(); console.log();
+/*********************************************************************************
+*                                   EXERCISE I                                   *
+**********************************************************************************/ 
+logHeader('EXERCISE 1', 'getLength(items, cb)');
 
-  // Potential Solution:
+/* getLength passes the length of the array into the callback. */
 
-  // Higher order function using "cb" as the call back
-  function firstItem(arr, cb) {
-    return cb(arr[0]);
-  }
+function getLength(arr, cb) { 
+  cb(arr.length);
+}
 
-  // Function invocation 
-  firstItem(items, function(first) {
-    console.log(first)
-  });
+getLength(items, function (length) {     
+  console.log('Output:\n');
+  console.log(length);
+  console.log(); // line break
```

Comment:
Remember, you can do something like this:

```js
function consoleLog(input) {
  console.log('Output:\n');
  console.log(input);
  console.log(); // line break
}

getLength(items, consoleLog);
last(items, consoleLog);
sumNums(1, 2, consoleLog);
```

I have a more involved spiel, but looking at your `logHeader` function, _I think you get it already_. 😂